### PR TITLE
Update tutorial to use buidler plugin

### PR DESCRIPTION
**Pull request by 0xGabi**
Depends on https://github.com/aragon/your-first-aragon-app/pull/14

It updates the tutorial a bit to the new standards we are using:

- React app with hooks
- Use `react-api` to fetch `appState`
- Use new aragonUI components
- Use the buidler plugin and a basic example of script hooks.

### CODE REVIEW

**Comment by 0xGabi** [comment]
Would you mind take a look @bpierre? Thanks

**Comment by 0xGabi** [comment]
No worries, we already test it with Mathew. Thanks anyways 🙌

File: docs/tutorial.md [hunk 1]
```diff
@@ -6,26 +6,61 @@ sidebar_label: Your first Aragon app
...
+```
 
-Notice that we input a fully qualified [ENS](https://ens.domains/) name. Let's examine the ENS name we entered, because it is not entirely arbitrary.
+- **app**: Frontend folder. Completely encapsulated, has its own package.json and dependencies.
+  - **src**: Source files.
+    - `App.js`: Aragon app root component.
+    - `index.js`: Aragon app entry point.
+    - `script.sol`: Aragon app background script.
+  - [**package.json**](https://docs.npmjs.com/creating-a-package-json-file): Frontend npm configuration file.
+- **contracts**: Smart Constracts folder.
```
Comment (sohkai) [inline docs/tutorial.md:49]: > Smart **contracts** folder

We may want to double check the other boilerplate repos and see if we also need to update them there.

File: docs/tutorial.md [hunk 2]
```diff
@@ -6,26 +6,61 @@ sidebar_label: Your first Aragon app
...
+- **scripts**: Scripts folder.
+  - `buidler-hooks.js`: Buidler script hook.
+- **test**: Tests folder.
+- [**arapp.json**](https://hack.aragon.org/docs/cli-global-confg#the-arappjson-file): Aragon configuration file. Includes Aragon-specific metadata for your app.
+- [**manifest.json**](https://hack.aragon.org/docs/cli-global-confg#the-manifestjson-file): Aragon configuration file. Includes web-specific configurations.
+- [**buidler.config.js**](https://buidler.dev/config/): Buidler configuration file.
+- [**package.json**](https://docs.npmjs.com/creating-a-package-json-file): Main npm configuration file.
 
-![Illustration of foo.aragonpm.eth](https://i.imgur.com/MQnYT6d.png)
+### Stuck?
```
Comment (sohkai) [inline docs/tutorial.md:59]: Might be a good place to add a link to Spectrum chat too

File: docs/tutorial.md [hunk 3]
```diff
@@ -140,125 +187,190 @@ Then the client takes care of connecting to Ethereum via Web3, and also handles
...
-
-Internally, `store` will fetch the last known state (if any) and pass that in as the first argument, and then store the resulting state in cache. This state can be observed in the view portion of your app. Also note that the `store` method returns an observable of states. This is a recurring theme in the JavaScript implementation of aragonAPI—almost everything is an [RxJS](http://reactivex.io/rxjs/) observable.
+The `store` method takes in a reducer function with the signature `(state, event) => state`, where `state` is whatever you want it to be (in this example it is an integer), and `event` is a [Web3 event](https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html#contract-events). The reducer function **must always** return a state, even if it is the same state as before. Returning undefined will reset the reduced state to its initial null state. Also note that the initial state is always null, not undefined, because of JSONRPC limitations.
```
Comment (sohkai) [inline docs/tutorial.md:258]: If we haven't already, adding a link to the `store()` API might be good.

File: docs/tutorial.md [hunk 4]
```diff
@@ -140,125 +187,190 @@ Then the client takes care of connecting to Ethereum via Web3, and also handles
...
 
-> **Note**<br>
-> The state will start out as `null`, not `undefined` because of the JSONRPC specification.
+The store has block caching automatically applied, such that subsequent loads of the application only fetch new events from a cached ("committed") block height (rather than from 0 or the app's initialization block). This state can be observed in the view portion of your app. Also note that the `store` method returns an observable of states. This is a recurring theme in the JavaScript implementation of aragonAPI—almost everything is an [RxJS](http://reactivex.io/rxjs/) observable.
 
 ### Displaying State
 
-Now let's write the view portion of our app. In our case, this is a simple HTML file, and a simple JavaScript file that observes the state that our background worker builds for us.
+Now let's write the view portion of our app. In our case, this is a simple HTML file, and a simple React app wiht the `useAragonApi` [React Hook](https://reactjs.org/docs/hooks-intro.html) that observes the state that our background script builds for us and returns the data needed to interact with the app contract.
```
Comment (sohkai) [inline docs/tutorial.md:266]: > React app **with**

**Pull request merged**